packages/typespec-azure-resource-manager/src/rules/core-operations.ts

```diff
@@ -31,7 +31,8 @@ export const coreOperationsRule = createRule({
       operation: (operation: Operation) => {
         if (
           !isInternalTypeSpec(context.program, operation) &&
-          !isSourceOperationResourceManagerInternal(operation)
+          !isSourceOperationResourceManagerInternal(operation) &&
+          !isArmProviderActionOperation(operation)
```

Comment:
I don't think we want to use this check here, for example, the `opOutsideInterface` check should still apply - this should narrowly prevent usage of the arm-resource-operation rule.

Also, I don't think we want to make exceptiosn to the rule this way - this should be about allowing provider actions,  meaning a static segment *after* the provider name in the operation route.  operations outside this, no matter what template they use should still be flagged.